Block/Product/View/Type/Configurable.php

```diff
@@ -0,0 +1,206 @@
+     * @param CurrentCustomer $currentCustomer
+     * @param PriceCurrencyInterface $priceCurrency
+     * @param ConfigurableAttributeData $configurableAttributeData
+     * @param SwatchData $swatchHelper
+     * @param Media $swatchMediaHelper
+     * @param array $data other data
+     * @param SwatchAttributesProvider $swatchAttributesProvider
+     * @SuppressWarnings(PHPMD.ExcessiveParameterList)
+     */
+    public function __construct(
+        Context $context,
+        ArrayUtils $arrayUtils,
+        EncoderInterface $jsonEncoder,
+        Data $helper,
+        CatalogProduct $catalogProduct,
+        CurrentCustomer $currentCustomer,
+        PriceCurrencyInterface $priceCurrency,
+        ConfigurableAttributeData $configurableAttributeData,
+        SwatchData $swatchHelper,
+        Media $swatchMediaHelper,
+        array $data = [],
+        SwatchAttributesProvider $swatchAttributesProvider = null,
+        Format $localeFormat = null,
+        StockItemInterfaceFactory $stockItemInterfaceFactory
+    ){
+        $this->swatchHelper = $swatchHelper;
+        $this->swatchMediaHelper = $swatchMediaHelper;
+        $this->_stockRepository     = $stockItemInterfaceFactory->create();
+        $this->swatchAttributesProvider = $swatchAttributesProvider
+            ?: ObjectManager::getInstance()->get(SwatchAttributesProvider::class);
+        $this->localeFormat = $localeFormat ?: ObjectManager::getInstance()->get(Format::class);
+
+        parent::__construct(
+            $context,
+            $arrayUtils,
+            $jsonEncoder,
+            $helper,
+            $catalogProduct,
+            $currentCustomer,
+            $priceCurrency,
+            $configurableAttributeData,
+            $swatchHelper,
+            $swatchMediaHelper,
+            $data
+        );
+    }
+
+    /**
+     * Sets product stock status.
+     *
+     * @param   $productId
+     *
+     * @return  array
+     */
+    public function getStockItem($productId)
+    {
+        $stock_data                   = array();
+        $stock_data['out_stock']      = 0 ;
+
+        $stock = $this->_stockRepository->load($productId,'product_id');
+        if (!$stock->getIsInStock()) {
+            $stock_data['out_stock'] = 1;
+        }
+
+        return $stock_data;
+    }
+
+    /**
+     * Get Product Stock
+     *
+     * @return array
+     */
+    public function getProductStock()
+    {
+        $stock = [];
+        $skipSaleableCheck=true;
+        $allProducts       = $this->getProduct()->getTypeInstance()->getUsedProducts($this->getProduct(), null);
+
+        foreach ($allProducts as $product) {
+            if ($product->isSaleable() || $skipSaleableCheck) {
+                $stock[$product->getId()] = $this->getStockItem($product->getId());
+            }
+        }
+        return $stock;
+    }
+
+
+    /**
+     * Composes configuration for js
+     *
+     * @return string
+     */
+    public function getJsonConfig()
```

Comment:
Is it necessary to override this function instead of operating on the parents return value?

Reply:
regarding getJsonConfig(), i think that is the place to add the values in to the json. i need to have a look if i could achieve that any other way.